test/nearleyc.test.js

```diff
@@ -207,3 +176,48 @@ describe('builtins', () => {
     })
 })
 
+describe('nearleyc: macros', () => {
+
+    it('seems to work', () => {
+        // Matches "'Hello?' 'Hello?' 'Hello?'"
+        const grammar = compile(`
+            matchThree[X] -> $X " " $X " " $X
+            inQuotes[X] -> "'" $X "'"
+            main -> matchThree[inQuotes["Hello?"]]
+        `);
+
+        expect(prettyPrint(grammar)).toEqual([
+            'main$macrocall$2$macrocall$2$string$1 → "H" "e" "l" "l" "o" "?"',
+            'main$macrocall$2$macrocall$2 → main$macrocall$2$macrocall$2$string$1',
+            'main$macrocall$2$macrocall$1 → "\'" main$macrocall$2$macrocall$2 "\'"',
+            'main$macrocall$2 → main$macrocall$2$macrocall$1',
+            'main$macrocall$1 → main$macrocall$2 " " main$macrocall$2 " " main$macrocall$2',
+            'main → main$macrocall$1',
+        ]);
+    });
+
+    it('must be defined before use', () => {
+        expect(() => compile(`
+            main -> matchThree[inQuotes["Hello?"]]
+            matchThree[X] -> $X " " $X " " $X
+            inQuotes[X] -> "'" $X "'"
+        `)).toThrow();
+    });
+
+    it('compiles a simple macro from external file', function() {
+        const grammar = compile(read("test/grammars/macro-test.ne"));
```

Comment:
Please could we use a template string `\`...\`` instead of `read()`? I'd rather bring the grammars into the test files themselves where possible (i.e. they're not too long).

Reply:
Well, part of the reason I added the test was actually to be able to reuse it for Lark and it needs to be an external file for that. https://github.com/erezsh/lark/issues/26

Reply:
Huh, fair enough.